warehouse/models/staging/payments/elavon/int_elavon__billing_transactions_deduped.sql

```diff
@@ -0,0 +1,79 @@
+{{ config(materialized='table') }}
+
+WITH
+
+stg_elavon__transactions AS (
+
+  SELECT * FROM {{ ref('stg_elavon__transactions') }}
+
+),
+
+billing_transactions AS (
+
+  SELECT * FROM stg_elavon__transactions
+  WHERE batch_type = 'B'
+
+),
+
+dedup_billing_transactions AS (
+
+  SELECT *
+  FROM billing_transactions
+  WHERE execution_ts = (SELECT MAX(execution_ts) FROM billing_transactions)
```

Comment:
I am a bit confused about why this one is different from the deposits one. For billing transactions do we only want things from the latest execution, but for deposits we want latest by `trn_ref_num`? And same as the other one it would be clearer to me as a `QUALIFY` statement, we have generally not really used subqueries elsewhere in our codebase but I can just deal if they're preferred in payments for whatever reason.

Reply:
So they're different because billing transactions don't have a `trn_ref_num` associated with them, so the only thing we can dedupe on is the latest execution. this is the heart of the issue with reconciling Elavon and Littlepay, there's no `trn_ref_num` to join on when there should be

Reply:
I agree with the `QUALIFY` statement though, I can implement that

Reply:
That makes sense, though I thought that we were using `purch_id` to map to Littlepay rather than `trn_ref_num`. Thank you for explaining!

Reply:
sorry yes you're right, purch_id is used for the mapping and isn't present in billing transactions, along with trn_ref_num and a slew of other fields